packages/shared/lib/services/connection.service.ts

```diff
@@ -113,6 +114,68 @@ class ConnectionService {
         return [{ connection: connection[0]!, operation: 'creation' }];
     }
 
+    public async upsertTbaConnection({
+        connectionId,
+        providerConfigKey,
+        credentials,
+        connectionConfig,
+        metadata,
+        config,
+        environment,
+        account
+    }: {
+        connectionId: string;
+        providerConfigKey: string;
+        credentials: TbaCredentials;
+        connectionConfig: ConnectionConfig;
+        config: ProviderConfig;
+        metadata: Metadata;
+        environment: Environment;
+        account: Account;
+    }): Promise<ConnectionUpsertResponse[]> {
+        const storedConnection = await this.checkIfConnectionExists(connectionId, providerConfigKey, environment.id);
+
+        if (storedConnection) {
+            const encryptedConnection = encryptionManager.encryptConnection({
+                connection_id: connectionId,
+                config_id: config.id as number,
+                provider_config_key: providerConfigKey,
+                metadata,
+                credentials,
+                connection_config: connectionConfig,
+                environment_id: environment.id
+            });
+            encryptedConnection.updated_at = new Date();
+            const connection = await db.knex
+                .from<StoredConnection>(`_nango_connections`)
+                .where({ id: storedConnection.id, deleted: false })
+                .update(encryptedConnection)
+                .returning('*');
+
+            void analytics.track(AnalyticsTypes.TBA_CONNECTION_INSERTED, account.id, { provider: config.provider });
+
+            return [{ connection: connection[0]!, operation: 'override' }];
+        }
+        const connection = await db.knex
+            .from<StoredConnection>(`_nango_connections`)
+            .insert(
+                encryptionManager.encryptConnection({
+                    connection_id: connectionId,
+                    provider_config_key: providerConfigKey,
+                    config_id: config.id as number,
+                    credentials,
+                    metadata,
+                    connection_config: connectionConfig,
+                    environment_id: environment.id
+                })
+            )
+            .returning('*');
```

Comment:
no explicit error handling. What is returned to the end user if any of the db queries fails

Reply:
This is the same logic (for better or worse) of the other upsert connections, so error handling should be added to all if anything, but given a time crunch will leave as is for now.